lib/Bindings/Python/circt/design_entry.py

```diff
@@ -0,0 +1,78 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from circt.dialects import hw
+
+import mlir.ir
+
+
+class Output:
+  __slots__ = [
+      "type",
+      "value"
+  ]
+
+  def __init__(self, type: mlir.ir.Type):
+    self.type = type
+    self.value = None
+
+  def set(self, val):
```

Comment:
More a note to self than anything, rather than exposing `output.set(...)`, this could use the generalized `connect(output, ...)` syntax when it exists.